plugins/pdr_deterministic_plugin/ufm_sim_web_service/isolation_mgr.py

```diff
@@ -448,13 +450,26 @@ def check_temp_issue(self, port_obj, row, timestamp):
             return None
         cable_temp = get_counter(Constants.TEMP_COUNTER, row, default=None)
         if cable_temp is not None and not pd.isna(cable_temp):
-            if cable_temp in ["NA", "N/A", "", "0C"]:
+            if cable_temp in ["NA", "N/A", "", "0C", "0"]:
                 return None
-            cable_temp = int(cable_temp.split("C")[0]) if type(cable_temp) == str else cable_temp
-            dT = abs(port_obj.counters_values.get(Constants.TEMP_COUNTER, 0) - cable_temp)
+            # Get new and saved temperature values
+            cable_temp = int(cable_temp.split("C")[0]) if isinstance(cable_temp, str) else cable_temp
+            old_cable_temp = port_obj.counters_values.get(Constants.TEMP_COUNTER);
+            # Save new temperature value
             port_obj.counters_values[Constants.TEMP_COUNTER] = cable_temp
-            if cable_temp and (cable_temp > self.tmax or dT > self.d_tmax):
+            # Check temperature condition
+            if cable_temp > self.tmax:
+                self.logger.info(f"Isolation issue ({Constants.ISSUE_OONOC}) triggered for port {port_obj.port_name}: "
+                                 f"cable temperature ({cable_temp}) is higher than TMAX ({self.tmax})")
                 return Issue(port_obj.port_name, Constants.ISSUE_OONOC)
+            # Check temperature delta condition
+            if old_cable_temp is not None and old_cable_temp != 0:
+                delta_temp = abs(old_cable_temp - cable_temp)
+                if delta_temp > self.d_tmax:
+                    self.logger.info(f"Isolation issue ({Constants.ISSUE_OONOC}) triggered for port {port_obj.port_name}: "
```

Comment:
"Issue detected" - we don't know yet if it will be triggered

Reply:
Fixed